### ISSUE TITLE
(Un)serializing dispatched actions

### ISSUE THREAD

**Issue by bjyoungblood** [goatslacker/alt#72]
I was intrigued by the `DispatcherRecorder`, so I went about trying build a debugging tool that would take a snapshot and then record some actions. I would like to be able to serialize the snapshot and events so they could be shared and replayed later or on another machine.

However, in order for this to work, I have to modify [this line](https://github.com/goatslacker/alt/blob/master/src/alt.js#L351) to use `Symbol.for()` instead of `Symbol()` so that I can replay the events after unserializing.

I see why you're not using `Symbol.for` to avoid conflicts in action classes/names, but I think it could be useful to provide a way to use `Symbol.for`.

I'm imagining something like:

```
const alt = new Alt({ exposeActions : true });
```

or for each action class
```
class MyActions { ... }
alt.createActions(MyActions, exports, { expose : true });
```

***

Perhaps there's another way to achieve this without modifying alt. Here's my [work-in-progress](https://github.com/bjyoungblood/alt-step-debugger/blob/master/components/recorder.jsx) (apologies for the spaghetti code -- it's getting a bit late for me).

**Comment by goatslacker** [comment]
This is great, I had the same exact thoughts a couple of days ago but forgot to create a ticket about it. I'm not sure what the implementation details should be but this should definitely be a thing.

Excited about your WIP.

**Comment by goatslacker** [comment]
I switched to Symbol.for in an experimental branch which I'll push later today. All the tests pass because I'm generating a UID each time :)

It also makes this possible using `Symbol.keyFor`

**Comment by goatslacker** [comment]
#77

**Comment by goatslacker** [comment]
And merged c18b7f2c926a9f7da5a785ec67a4e930406a76a0
